dvc_azure/__init__.py

```diff
@@ -174,10 +120,13 @@ def fs(self):
 
         from .spec import AzureBlobFileSystem
 
+        # Disable spam from failed cred types for DefaultAzureCredential
+        logging.getLogger("azure.identity.aio").setLevel(logging.ERROR)
+
```

Comment:
Again better to do this here at runtime than in `_prepare_credentials`, since the latter doesn't even mean we will be using the filesystem.